synapse/handlers/space_summary.py

```diff
@@ -144,12 +200,25 @@ async def _summarize_local_room(
             )
         return (room_entry,), events_result
 
-    async def _is_room_accessible(self, room_id: str, requester: str) -> bool:
-        try:
-            await self._auth.check_user_in_room_or_world_readable(room_id, requester)
+    async def _is_room_accessible(self, room_id: str, requester: Optional[str]) -> bool:
+        # if we have an authenticated requesting user, first check if they are in the
+        # room
+        if requester:
+            try:
+                await self._auth.check_user_in_room(room_id, requester)
+                return True
+            except AuthError:
+                pass
+
+        # otherwise, check if the room is peekable
+        hist_vis = ""
+        hist_vis_ev = await self._state_handler.get_current_state(
+            room_id, EventTypes.RoomHistoryVisibility, ""
+        )
+        if hist_vis_ev:
+            hist_vis = hist_vis_ev.content.get("history_visibility")
+        if hist_vis == HistoryVisibility.WORLD_READABLE:
             return True
```

Comment:
I think it would be clearer to indent this since it can only be reached if the event exists.

Reply:
true.